### PR TITLE
Feat/aapd 1542 draft appeals on dashboard

### PR DESCRIPTION
**Pull request by DeanH1990**
## Ticket Number

https://pins-ds.atlassian.net/browse/AAPD-1542

## Description of change

gets and displays v2 appellant submissions on the dashboard display. 

## Checklist

- [ ] Requires infrastructure changes
- [ ] I have updated the documentation accordingly
- [x] My commit history in this PR is linear
- [ ] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `main` (please only [rebase](https://github.com/Planning-Inspectorate/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: packages/database/src/seed/data-dev.js [hunk 1]
```diff
@@ -719,22 +733,48 @@ const neighbourAddresses = [
...
 		applicationDecisionDate: new Date(),
 		Appeal: {
 			connect: { id: appealIds.appealNine }
 		}
+	},
+	{
+		id: appellantSubmissionIds.appellantSubmissionThree,
+		LPACode: 'Q9999',
+		appealTypeCode: 'HAS',
+		applicationDecisionDate: new Date('2024-03-01'),
```
Comment (DeanH1990) [inline packages/database/src/seed/data-dev.js:758]: specified date here to bring deadline to a few days away

File: dev/data/data/appeals-service-api/003-appeals.json [hunk 2]
```diff
@@ -99,6 +99,106 @@
...
+			"aboutYouSection": {
+				"yourDetails": {
+					"appealingOnBehalfOf": "",
+					"email": "bob@smith.com",
+					"name": "Bob Smith",
+					"isOriginalApplicant": true
+				}
+			},
+			"state": "DRAFT",
+			"decisionDate": "2024-03-01T12:00:00Z",
```
Comment (DeanH1990) [inline dev/data/data/appeals-service-api/003-appeals.json:196]: both these dates mean the deadline is 11pm friday.
Can move it back one day so they disappear thursday eve for testers to confirm they dont show on friday?

Comment (bryn-pins) [reply dev/data/data/appeals-service-api/003-appeals.json:196]: ok so I'm remembering after looking at this that I don't think these actually get populated to the environments and is a local only thing, can copy it manually into mongo so will do that, once this goes through

File: packages/appeals-service-api/src/routes/v2/users/_id/appeals/service.js [hunk 3]
```diff
@@ -33,6 +33,13 @@ async function getAppealsForUser(userId) {
 		.map((appealToUser) => appealToUser.Appeal?.legacyAppealSubmissionId)
 		.filter(filterNotNull);
 
+	// find v2 draft submissions
+	const v2Drafts = user.Appeals.filter(
+		(appealToUser) =>
+			appealToUser.Appeal?.AppellantSubmission?.submitted === false &&
+			appealToUser.Appeal?.AppealCase === null
```
Comment (bryn-pins) [inline packages/appeals-service-api/src/routes/v2/users/_id/appeals/service.js:40]: do we need the second line? won't submitted be enough?

File: packages/business-rules/src/constants.js [hunk 4]
```diff
@@ -13,6 +13,11 @@ const APPEAL_ID = {
 	FAST_TRACK_TREES: '1011'
 };
 
+const TYPE_CODE = {
+	HOUSEHOLDER: 'HAS',
```
Comment (bryn-pins) [inline packages/business-rules/src/constants.js:17]: think these exist in common/src/constants already

File: packages/appeals-service-api/src/routes/v2/users/_id/appeals/service.js (hunk 3 above)
Comment (DeanH1990) [reply packages/appeals-service-api/src/routes/v2/users/_id/appeals/service.js:40]: Yes you're correct. 

I had added the second line as the first appeal submission that was added when first building the journey was attached to an appeal that had the appeal case published. Obviously going through the complete journey normally this wouldn't happen.

Ive made that first submission draft as submitted = true and removed the appeal case check.

File: packages/business-rules/src/constants.js (hunk 4 above)
Comment (DeanH1990) [reply packages/business-rules/src/constants.js:17]: Have changed to use the ones in common.

Will push up in just a min

File: dev/data/data/appeals-service-api/003-appeals.json (hunk 2 above)
Comment (DeanH1990) [reply dev/data/data/appeals-service-api/003-appeals.json:196]: Dont think I have access to Mongo. But happy to add it if i do, when its gone through

**Review by bryn-pins — APPROVED** [review]

**Pull request merged**